lib/matplotlib/artist.py

```diff
@@ -1656,3 +1709,7 @@ def kwdoc(artist):
     return ('\n'.join(ai.pprint_setters_rest(leadingspace=4))
             if mpl.rcParams['docstring.hardcopy'] else
             'Properties:\n' + '\n'.join(ai.pprint_setters(leadingspace=4)))
+
+# We defer this to the end of them module, because it needs ArtistInspector
+# to be defined.
+Artist._update_set_signature_and_docstring()
```

Comment:
Can you just move the definition of ArtistInspector above Artist and avoid this deferred call? (And then _update_set_signature_and_docstring can be inlined into init_subclass.)

Reply:
No strong opinion. But that's quite a bit moving of code, which makes digging in the history more complicated.

I don't think we can inline, because `__init_subclass__` is not called by `Artist` itself, but we want to adjust `Artist.set` as well. (I wouldn't want to explicitly call `__init_subclass__` on `Artist`. That's more confusing than an extra function.)

Reply:
Let's leave things as they are for now.